tests/components/sensor/test_snmp.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for sensor.snmp."""
+
+import unittest
+from homeassistant.setup import setup_component
+from tests.common import get_test_home_assistant
+from threading import Thread
+
+from pysnmp.entity import engine, config
+from pysnmp.entity.rfc3413 import cmdrsp, context
+from pysnmp.carrier.asyncore.dgram import udp
+from pysnmp.proto.api import v2c
+from pysnmp.proto.rfc1902 import Opaque
+
+import time
+
+_PORT = 10161
+_BASE_OID = (1, 3, 6, 1, 4, 1, 6574, 4)
+_OID1 = _BASE_OID + (1, 1, 1)
+_OID2 = _BASE_OID + (2, 1, 1)
+_OID3 = _BASE_OID + (3, 1, 1)
+
+
+class TestSnmp(unittest.TestCase):
+    """Test basic functionality of sensor.snmp."""
+
+    def setUp(self):
+        """Setup a hoss instance for testing."""
+        self.hass = get_test_home_assistant()
+        self.hass.start()
+        self.snmpEngine = None
+        self._agent_thread = None
+
+    def tearDown(self):
+        """Clean up after the tests.
+
+        Especially take care of the snmpEngine.
+        """
+        if self.snmpEngine is not None:
+            self.snmpEngine.transportDispatcher.jobFinished(1)
+            self.snmpEngine.transportDispatcher.unregisterRecvCbFun(
+                recvId=None)
+            self.snmpEngine.transportDispatcher.unregisterTransport(
+                udp.domainName)
+        if self._agent_thread is not None:
+            self._agent_thread.join(0.01)
+        self.hass.stop()
+
+    def _run_agent(self):
+        """Run a snmp thread so that we can get data from it.
+
+        This is based on the examples from the pysnmp library.
+        """
+        self.snmpEngine = engine.SnmpEngine()
+        config.addTransport(
+            self.snmpEngine,
+            udp.domainName,
+            udp.UdpTransport().openServerMode(('127.0.0.1', _PORT))
```

Comment:
I think the engine should be mocked. Why do we need a real engine? We only need to test the home assistant side of the logic.